ngraph/frontend/onnx_import/src/op/log_softmax.cpp

```diff
@@ -17,32 +17,95 @@
 #include <memory>
 
 #include "log_softmax.hpp"
+#include "ngraph/builder/reshape.hpp"
 #include "ngraph/validation_util.hpp"
 #include "onnx_import/default_opset.hpp"
 
 namespace ngraph
 {
     namespace onnx_import
     {
-        namespace op
+        namespace detail
         {
-            namespace set_1
+            std::shared_ptr<ngraph::Node> onnx_logsoftmax(const Output<ngraph::Node> data,
+                                                          const int64_t axis)
             {
-                OutputVector log_softmax(const Node& node)
+                const auto coerced_data = ngraph::builder::opset1::flatten(data, axis);
+
+                const auto axis_1 = default_opset::Constant::create(element::i64, Shape{1}, {1});
+                const auto max = std::make_shared<default_opset::ReduceMax>(coerced_data, axis_1);
+
+                // equivalent to numpy's max.reshape((-1,1))
+                const auto reshape_pattern =
+                    default_opset::Constant::create(element::i64, Shape{2}, {0, 1});
+                const auto reshaped_max =
+                    std::make_shared<default_opset::Reshape>(max, reshape_pattern, true);
+
+                const auto data_minus_max =
+                    std::make_shared<default_opset::Subtract>(coerced_data, reshaped_max);
```

Comment:
Could you check if this change will work for you?
```suggestion
                const auto max = std::make_shared<default_opset::ReduceMax>(coerced_data, axis_1, true);

                const auto data_minus_max =
                    std::make_shared<default_opset::Subtract>(coerced_data, max);
```